src/Microsoft.AspNetCore.Mvc.Razor/DependencyInjection/MvcRazorMvcCoreBuilderExtensions.cs

```diff
@@ -116,6 +152,11 @@ internal static void AddRazorViewEngineServices(IServiceCollection services)
                 return new DefaultChunkTreeCache(accessor.FileProvider);
             }));
 
+            services.TryAddTransient<ITagHelperDescriptorResolver, TagHelperDescriptorResolver>();
+
+            services.TryAddTransient<ITagHelperTypeResolver, TagHelperTypeResolver>();
+            services.TryAddTransient<ITagHelperDescriptorFactory>(s => new TagHelperDescriptorFactory(designTime: false));
```

Comment:
We might want to a add a parameterless constructor to TagHelperDescriptorFactory that calls into TagHelperDescriptorFactory(bool) with design time = false for use on DI and we can remove the use of the factory.

Also, these dependencies can be made singletons